app/src/main/java/de/markusfisch/android/shadereditor/widget/ShaderEditor.java

```diff
@@ -300,13 +311,13 @@ public void afterTextChanged(Editable e) {
 				}
 
 				dirty = true;
+				highlightWithoutChange(e);
```

Comment:
I haven't tested it yet, but I have a guess. I have changed the highlighting to be instantaneous, because the Lexer can work in realtime. Adding spans does take some time, that's why I added the `diff` method. I think this could be the root of the performance regression

Reply:
Hm, yes, that may play a role indeed 🤔